commons/src/main/java/org/infinispan/commons/equivalence/EntryEquivalence.java

```diff
@@ -0,0 +1,62 @@
+package org.infinispan.commons.equivalence;
+
+import java.util.Map;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class EntryEquivalence<K, V> implements Equivalence<Map.Entry<K, V>> {
```

Comment:
I don't think this is necessary, considering that `Map.entrySet()` isn't supposed to let the user add elements.